syncgroup.go

```diff
@@ -37,6 +37,10 @@ func (t groupAssignment) writeTo(w *bufio.Writer) {
 }
 
 func (t *groupAssignment) readFrom(r *bufio.Reader, size int) (remain int, err error) {
+	if size == 0 {
```

Comment:
Are there valid use cases where there will be no more bytes left to read even tho we expect to read a `groupAssignment` structure?

Reply:
I came across this case when testing for compatibility with bsm/sarama-cluster.  It appears in some cases, sarama-cluster can send a nil array entry.  Admittedly, I didn't look too closely at it.

Reply:
How about leaving a comment about it here? So anyone reading this code knows why we’re making the check.

Reply:
Sure, that makes sense.